core-rust/state-manager/src/receipt.rs

```diff
@@ -142,16 +149,17 @@ pub struct LocalTransactionExecution {
 
 impl LedgerTransactionReceipt {
     pub fn get_consensus_receipt(&self) -> ConsensusReceipt {
+        let substate_change_hashes = self
+            .substate_changes
+            .iter()
+            .map(|substate_change| scrypto_encode(substate_change).unwrap())
```

Comment:
The `SubstateChange.get_hash()` (for getting the hash for substate change merkle tree) is an excellent idea, very natural 👍 

It is a bit more arbitrary about the `SubstateHashChange`.
_(wow, I didn't realize these 2 names are so close to each other [in terms of Levenshtein] :( But at the same time they are so accurate... [in one case, we hash the substate change, and in the other, we capture a change of the substate hash 🤷 ])_
Anyway, for getting the change of the substate hash (for state JMT), we actually don't care about hashing substate id, or a previous substate value... It seems we want just the `ChangeAction.get_new_value_hash() -> Option<Hash>`, so that's what I added 👍